modules/python/python-logtemplate.c

```diff
@@ -0,0 +1,163 @@
+{
+  PyLogTemplate *self = (PyLogTemplate *)s;
+
+  PyLogMessage *msg;
+  PyLogTemplateOptions *py_log_template_options = NULL;
+  gint tz = LTZ_SEND;
+  gint seqnum = 0;
+
+  static const gchar *kwlist[] = {"msg", "options", "tz", "seqnum", NULL};
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwrds, "O|Oii", (gchar **)kwlist,
+                                   &msg, &py_log_template_options, &tz, &seqnum))
+    return NULL;
+
+  if (!py_is_log_message((PyObject *)msg))
+    {
+      PyErr_Format(PyExc_TypeError,
+                   "LogMessage expected in the first parameter");
+      return NULL;
+    }
+
+  if (py_log_template_options && !py_is_log_template_options((PyObject *)py_log_template_options))
+    {
+      PyErr_Format(PyExc_TypeError,
+                   "LogTemplateOptions expected in the second parameter");
+      return NULL;
+    }
+
+  LogTemplateOptions *log_template_options = py_log_template_options ? py_log_template_options->template_options :
+                                             self->template_options;
+  if (!log_template_options)
+    {
+      PyErr_Format(PyExc_RuntimeError,
+                   "LogTemplateOptions must be provided either in the LogTemplate constructor or as parameter of format");
+      return NULL;
+    }
+
+  GString *result = scratch_buffers_alloc();
+  log_template_format(self->template, msg->msg, log_template_options, tz, seqnum, NULL, result);
+
+  return _py_string_from_string(result->str, result->len);
+}
+
+PyObject *
+py_log_template_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+  gchar *template_string;
+  PyLogTemplateOptions *py_log_template_options = NULL;
+  if (!PyArg_ParseTuple(args, "s|O", &template_string, &py_log_template_options))
+    return NULL;
+
+  if (py_log_template_options && !py_is_log_template_options((PyObject *)py_log_template_options))
+    {
+      PyErr_Format(PyExc_TypeError,
+                   "LogTemplateOptions expected in the second parameter");
+      return NULL;
+    }
+
+  LogTemplate *template = log_template_new(NULL, NULL);
+  GError *error = NULL;
+  if (!log_template_compile(template, template_string, &error))
+    {
+      PyErr_Format(PyExc_LogTemplate,
+                   "Error compiling template: %s", error->message);
+      g_clear_error(&error);
+      log_template_unref(template);
+      return NULL;
+    }
+
+  PyLogTemplate *self = (PyLogTemplate *)type->tp_alloc(type, 0);
+  if (!self)
+    {
+      log_template_unref(template);
+      return NULL;
+    }
+
+  self->template = template;
+  if (py_log_template_options)
+    self->template_options = py_log_template_options->template_options;
+
+  return (PyObject *)self;
+}
+
+static PyMethodDef py_log_template_methods[] =
+{
+  { "format", (PyCFunction)py_log_template_format, METH_VARARGS | METH_KEYWORDS, "format template" },
+  { NULL }
+};
+
+PyTypeObject py_log_template_type =
+{
+  PyVarObject_HEAD_INIT(&PyType_Type, 0)
```

Comment:
Shouldn't we use `PyObject_HEAD_INIT` here? PyLogTemplate is a `PyObject`, not a `PyVarObject`.